### PR TITLE
Machine class fixes (Darwin, *BSD)

### PR DESCRIPTION
**Pull request by Ximalas**
A series of changes to get rid of errors and warnings for FreeBSD and NetBSD, introduced by the new Machine class.

### CODE REVIEW

**Comment by natoscott** [comment]
@Ximalas can you check if this is a subset of #1239 please?

**Comment by Ximalas** [comment]
> @Ximalas can you check if this is a subset of #1239 please?

Yes, it is.

**Comment by natoscott** [comment]
OK, thanks, I'll squash & merge 'em as best I ca.

**Comment by BenBE** [comment]
@Ximalas Can you cherry-pick the NetBSD changes from #1239 (Updat the commit message accordingly)? Those are related to the build failures (introduced through changes from #1234). AFAICS the patch from there should apply without (major) collisions.

Regarding your patches: In `freebsd/FreeBSDProcessList.h` please move `freebsd/FreeBSDMachine.h` and `freebsd/FreeBSDProcess.h` (path missing) to below `XUtils.h`. Amend your patch.

Also in `freebsd/FreeBSDMachine.c` for `Machine_new` you can return `super` directly (instead of casting `this`.

@natoscott : I think this PR should superseed #1239 as your PR is missing some of the header changes included here.

**Comment by BenBE** [comment]
@Ximalas Can you cherry-pick the NetBSD changes from #1239 (Updat the commit message accordingly)? Those are related to the build failures (introduced through changes from #1234). AFAICS the patch from there should apply without (major) collisions.

Regarding your patches: In `freebsd/FreeBSDProcessList.h` please move `freebsd/FreeBSDMachine.h` and `freebsd/FreeBSDProcess.h` (path missing) to below `XUtils.h`. Amend your patch.

Also in `freebsd/FreeBSDMachine.c` for `Machine_new` you can return `super` directly (instead of casting `this`.

@natoscott : I think this PR should superseed #1239 as your PR is missing some of the header changes included here.

File: freebsd/FreeBSDProcessList.c [hunk 1]
```diff
@@ -7,6 +7,7 @@ in the source distribution for its full text.
 
 #include "config.h" // IWYU pragma: keep
 
+#include "freebsd/FreeBSDMachine.h"
```
Comment (BenBE) [inline freebsd/FreeBSDProcessList.c:10]: Header order, should go below `XUtils.h` as this is not paired with `FreeBSDMachine.c`.

On that note: `FreeBSDProcess.h` should be `freebsd/FreeBSDProcess.h` instead (sorted below `XUtils.h` too.

File: freebsd/FreeBSDMachine.c [hunk 2]
```diff
@@ -146,13 +146,13 @@ Machine* Machine_new(UsersTable* usersTable, uid_t userId) {
       CRT_fatalError("kvm_openfiles() failed");
    }
 
-   return this;
+   return (Machine *)this;
```
Comment (BenBE) [inline freebsd/FreeBSDMachine.c:149]: Return `super` to avoid cast.

**Review by BenBE — CHANGES_REQUESTED** [review]
Please include the NetBSD changes from #1239 as noted previously.

**Comment by BenBE** [comment]
I had a closer look at the other platforms too and found some more minor things to update for consistency and correctnes.

The commits are mostly squashed from yours, except for the additional changes for Darwin, DragonFlyBSD and OpenBSD. I picked the NetBSD changes from @natoscott and reworded the commit to reflect the new commit contents.

@Ximalas: Note that I updated your branch ref accordingly. Please follow up further changes from there.

File: freebsd/FreeBSDProcessList.c [hunk 3]
```diff
@@ -38,14 +37,17 @@ in the source distribution for its full text.
 #include "Settings.h"
 #include "XUtils.h"
 
+#include "freebsd/FreeBSDMachine.h"
+#include "freebsd/FreeBSDProcess.h"
+
 
 ProcessList* ProcessList_new(Machine* host, Hashtable* pidMatchList) {
```
Comment (BenBE) [inline freebsd/FreeBSDProcessList.c:44]: Following along with other platforms this should actually be called `FreeBSDProcessList_new`.

**Review by BenBE — APPROVED** [review]
I think only minor style issues left.

@natoscott: I think we should have a closer look at the `Machine_done` calls on the various platforms. I know, historically, we had the super class free all memory (fine so far), but in that cast we should also consider freeing the super class members before inherited members, i.e. `free(this->…);` before calling `Machine_done(super);` closed by `free(this);`. Nothing that blocks this PR, but an item for the overall refactoring list.

File: freebsd/FreeBSDProcessList.c (hunk 3 above)
Comment (natoscott) [reply freebsd/FreeBSDProcessList.c:44]: @BenBE that makes sense, but its not correct ;)  ...

```
grep ProcessList_new */*ProcessList.c
```

**Comment by natoscott** [comment]
> [...] also consider freeing the super class members before inherited members

+1  ... I'm pretty sure we're not consistent about this.  I'll add it to the cleanup queue, thanks @BenBE

**Review by natoscott — COMMENTED** [review]
LGTM, thanks all.  :)

**Pull request merged**